state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -366,28 +410,26 @@ pub mod pallet {
 		},
 		/// A swap has been executed.
 		SwapExecuted {
+			swap_request_id: SwapRequestId,
```

Comment:
if i understand things right, we would not be able to determine if this is the `gas` part of a `ccm` swap atm, right?

Reply:
i guess we could read the `gasBudget` from the `SwapRequested` and try to match that

Reply:
i assume the swap `swap_id` would correlate to the gas or principal swap id

Reply:
but i think we dont know the `swap_id` of the gas/principle swap (bc there is no event that announces them)?

Reply:
There is `SwapScheduled { swap_type, .. }` where swap_type is: 

```rust
pub enum SwapType {
	Swap,
	CcmPrincipal,
	CcmGas,
	NetworkFee,
	IngressEgressFee,
}
```

Reply:
you are right - missed that! can resolve this then